pkg/disk/v1alpha2/disk.go

```diff
@@ -0,0 +1,251 @@
+	var plist predicateList
+	plist = append(plist, p...)
+	if len(plist) == 0 {
+		return l
+	}
+
+	filtered := NewListBuilder().List()
+	for _, spcAPI := range l.ObjectList.Items {
+		spcAPI := spcAPI // pin it
+		Disk := BuilderForAPIObject(&spcAPI).Disk
+		if plist.all(Disk) {
+			filtered.ObjectList.Items = append(filtered.ObjectList.Items, *Disk.Object)
+		}
+	}
+	return filtered
+}
+
+// NewBuilder returns an empty instance of the Builder object.
+func NewBuilder() *Builder {
+	return &Builder{
+		Disk: &Disk{&apis.Disk{}},
+	}
+}
+
+// BuilderForObject returns an instance of the Builder object based on spc object
+func BuilderForObject(Disk *Disk) *Builder {
+	return &Builder{
+		Disk: Disk,
+	}
+}
+
+// BuilderForAPIObject returns an instance of the Builder object based on spc api object.
+func BuilderForAPIObject(spc *apis.Disk) *Builder {
+	return &Builder{
+		Disk: &Disk{spc},
+	}
+}
+
+// GetDeviceID returns the device ID of the disk.
+func (d *Disk) GetDeviceID() string {
+	var deviceID string
+	if len(d.Object.Spec.DevLinks) != 0 && len(d.Object.Spec.DevLinks[0].Links) != 0 {
+		deviceID = d.Object.Spec.DevLinks[0].Links[0]
+	} else {
+		deviceID = d.Object.Spec.Path
+	}
+	return deviceID
+}
+
+// Build returns the Disk object built by this builder.
+func (sb *Builder) Build() *Disk {
+	return sb.Disk
+}
+
+// NewListBuilder returns a new instance of ListBuilder object.
+func NewListBuilder() *ListBuilder {
+	return &ListBuilder{DiskList: &DiskList{ObjectList: &apis.DiskList{}}}
+}
+
+// WithList builds the list based on the provided *DiskList instances.
+func (b *ListBuilder) WithList(pools *DiskList) *ListBuilder {
+	if pools == nil {
+		return b
+	}
+	b.DiskList.ObjectList.Items = append(b.DiskList.ObjectList.Items, pools.ObjectList.Items...)
+	return b
+}
+
+// WithAPIList builds the list based on the provided *apis.CStorPoolList.
+func (b *ListBuilder) WithAPIList(pools *apis.DiskList) *ListBuilder {
+	if pools == nil {
+		return b
+	}
+	for _, pool := range pools.Items {
+		pool := pool //pin it
+		b.DiskList.ObjectList.Items = append(b.DiskList.ObjectList.Items, pool)
+	}
+	return b
+}
+
+// List returns the list of disk instances that were built by this builder.
+func (b *ListBuilder) List() *DiskList {
+	return b.DiskList
+}
+
+// NewListBuilderForObjectList builds the list based on the provided *DiskList instances.
+func NewListBuilderForObjectList(disk *DiskList) *ListBuilder {
+	newLB := NewListBuilder()
+	newLB.DiskList.ObjectList.Items = append(newLB.DiskList.ObjectList.Items, disk.ObjectList.Items...)
+	return newLB
+}
+
+// NewListBuilderForAPIList returns a new instance of ListBuilderForApiList object based on csp api list.
+func NewListBuilderForAPIList(diskAPIList *apis.DiskList) *ListBuilder {
+	newLb := NewListBuilder()
+	for _, obj := range diskAPIList.Items {
+		// pin it
+		obj := obj
+		newLb.DiskList.ObjectList.Items = append(newLb.DiskList.ObjectList.Items, obj)
```

Comment:
is it possible to avoid the for loop completely?
```go
newLb.DiskList.ObjectList.Items = append(newLb.DiskList.ObjectList.Items, diskAPIList.Items...)
```

Reply:
done